### ISSUE TITLE
turf.booleanPointOnLine() returning false even though the point is generated by turf.along()

### ISSUE THREAD

**Issue by zharfanzahisham** [Turfjs/turf#2390]
Hi, I am facing an issue where turf.booleanPointOnLine returns false even though the point is generated using turf.along. Steps to reproduce:

```javascript
var p1 = turf.along(line_geojson, 0.01);
turf.booleanPointOnLine(p1, line_geojson);  // false
```

Version: 6.5.0
line_geojson gist file: https://gist.github.com/zharfanzahisham/b988e927f75336fe1dbde390e577c739

Is there anything I'm doing wrong or missing? Thanks in advance!

**Comment by JamesLMilner** [comment]
Thanks for raising - I agree that the result of `along` should should probably pass `booleanPointOnLine`. As such I will flag this as a bug. Without investigating further it's hard to gauge which package is at fault here.

Here is the code as an example for anyone investigating:

https://codesandbox.io/s/zen-hooks-km5b4k?file=/src/index.ts

**Comment by RaymondDashWu** [comment]
@JamesLMilner I started investigating this and I think it's an issue of floating point comparisons? I took a look at [isPointOnLineSegment](https://github.com/Turfjs/turf/blob/d72985ce1a577b42340fed5fc70efe8e4bc8b062/packages/turf-boolean-disjoint/index.ts#L150), which is called by booleanPointOnLine. It's doing comparisons but I'm just not 100% sure what is being compared on L165 of that same file. I couldn't find a test for that particular function either. Would love your insight.
